src/components/button/button.tsx

```diff
@@ -200,7 +200,12 @@ export class Button
   }
 
   componentDidLoad(): void {
+    const { contentEl, childEl, fullLengthSlottedText } = this;
     setComponentLoaded(this);
+
+    contentEl.offsetWidth < contentEl.scrollWidth
```

Comment:
@Elijbet can you create an internal state for whether to set the title attribute? Maybe you can just reuse the `fullLengthSlottedText` something like so...

```ts
private handleSlotchange = (event): void => {
const {contentEl} = this;
    const textNodes = event.target.assignedNodes().filter((node) => node.nodeName === "#text");
    this.fullLengthSlottedText = contentEl.offsetWidth < contentEl.scrollWidth ? textNodes.map((node) => node.textContent).join("") : null;
  };
```

Then in the JSX you can just do this...

```jsx
<childEl
title={this.fullLengthSlottedText}
```

I think you will also still need to setup a resize observer on the childEl because if the text size changes after componentDidLoad you will need to run the logic whether to apply the title again. The size of the button could change due to other props as well so a resizeObserver is probably necessary.

Reply:
Can you try this?

```ts
this.fullLengthSlottedText = contentEl.offsetWidth < contentEl.scrollWidth ?
  contentEl.innerText : 
   null;
```

Reply:
Ah, ok. Now I understand.